File: Módulo Python/Códigos/6 - Funções/Exercicio.07.py
#7.	Faça um programa que receba 3 números e mostre, utilizando funções:
# -	Os números em ordem crescente
# -	Os números em ordem decrescente
# -	Os números pares
# -	Os números ímpares
# -	Os números maiores que 3 e menores que 10

n1 = float(input('Digite número 1: '))
n2 = float(input('Digite número 2: '))
n3 = float(input('Digite número 3: '))

def ordemCrescente(n1,n2,n3):
    if n1 <= n2 and n1 <= n3:
        menor = n1
        if n2 <= n3:
            medio = n2
            maior = n3
        else:
            medio = n3
            maior = n2
    else:
        if n2 <= n1 and n2 <= n3:
            menor = n2
            if n1 <= n3:
                medio = n1
                maior = n3
            else:
                medio = n3
                maior = n1
        else:
            menor = n3
            if n1 <= n2:
                medio = n1
                maior = n2
            else:
                medio = n2
                maior = n1

    print(f'Os números em ordem crescente são: {menor}, {medio}, {maior}')

def ordemDecrescente (n1, n2, n3):
    if n1 >= n2 and n1 >= n3:
        maior = n1
        if n2 >= n3:
            medio = n2
            menor = n3
        else:
            medio = n3
            menor = n2
    else:
        if n2 >= n1 and n2 >= n3:
            maior = n2
            if n1 >= n3:
                medio = n1
                menor = n3
            else:
                medio = n3
                menor = n1
        else:
            maior = n3
            if n1 >= n2:
                medio = n1
                menor = n2
            else:
                medio = n2
                menor = n1

    print(f'Os números em ordem decrescente são: {maior}, {medio}, {menor}')

def imparPar(n1, n2, n3):
    numeros = [n1, n2, n3]
    for num in numeros:
        if num % 2 == 0:
            print(f'O número {num} é par')
        else:
            print(f'O número {num} é ímpar')

def maiorMenor(n1, n2, n3):
    numeros = [n1, n2, n3]
    for num in numeros:
        if num > 3 and num < 10:
            print(f'O número {num} é maior que 3 e menor que 10')
        else:
            print(f'O número {num} não é maior que 3 e menor que 10')

ordemCrescente(n1,n2,n3)
ordemDecrescente(n1,n2,n3)
imparPar(n1, n2, n3)
maiorMenor(n1, n2, n3)

File: Livro - Módulo Java/6 - Funções/Exercicio.07.java
// 7 - Faça um programa que receba 3 números e mostre, utilizando funções: 
// - Os números em ordem crescente 
// - Os números em ordem decrescente 
// - Os números pares 
// - Os números ímpares 
// - Os números maiores que 3 e menores que 10 

import java.util.Arrays;
import java.util.Scanner;

public class AnaliseNumeros {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Digite o primeiro número:");
        int num1 = scanner.nextInt();

        System.out.println("Digite o segundo número:");
        int num2 = scanner.nextInt();

        System.out.println("Digite o terceiro número:");
        int num3 = scanner.nextInt();

        // Chamando as funções para realizar as diferentes análises
        imprimirOrdemCrescente(num1, num2, num3);
        imprimirOrdemDecrescente(num1, num2, num3);
        imprimirNumerosPares(num1, num2, num3);
        imprimirNumerosImpares(num1, num2, num3);
        imprimirNumerosEntre3e10(num1, num2, num3);

        scanner.close();
    }

    // Função para imprimir os números em ordem crescente
    public static void imprimirOrdemCrescente(int num1, int num2, int num3) {
        int[] numeros = {num1, num2, num3};
        Arrays.sort(numeros);
        System.out.println("Números em ordem crescente: " + Arrays.toString(numeros));
    }

    // Função para imprimir os números em ordem decrescente
    public static void imprimirOrdemDecrescente(int num1, int num2, int num3) {
        int[] numeros = {num1, num2, num3};
        Arrays.sort(numeros);
        for (int i = numeros.length - 1; i >= 0; i--) {
            System.out.print(numeros[i] + " ");
        }
        System.out.println("\nNúmeros em ordem decrescente.");
    }

    // Função para imprimir os números pares
    public static void imprimirNumerosPares(int num1, int num2, int num3) {
        System.out.print("Números pares: ");
        if (num1 % 2 == 0) System.out.print(num1 + " ");
        if (num2 % 2 == 0) System.out.print(num2 + " ");
        if (num3 % 2 == 0) System.out.print(num3 + " ");
        System.out.println();
    }

    // Função para imprimir os números ímpares
    public static void imprimirNumerosImpares(int num1, int num2, int num3) {
        System.out.print("Números ímpares: ");
        if (num1 % 2 != 0) System.out.print(num1 + " ");
        if (num2 % 2 != 0) System.out.print(num2 + " ");
        if (num3 % 2 != 0) System.out.print(num3 + " ");
        System.out.println();
    }

    // Função para imprimir os números maiores que 3 e menores que 10
    public static void imprimirNumerosEntre3e10(int num1, int num2, int num3) {
        System.out.print("Números entre 3 e 10: ");
        if (num1 > 3 && num1 < 10) System.out.print(num1 + " ");
        if (num2 > 3 && num2 < 10) System.out.print(num2 + " ");
        if (num3 > 3 && num3 < 10) System.out.print(num3 + " ");
        System.out.println();
    }
}